dashboard/test/controllers/api_controller_test.rb

```diff
@@ -1102,6 +1102,29 @@ class ApiControllerTest < ActionController::TestCase
     assert_response :forbidden
   end
 
+  test 'clever_classrooms queries clever with user uid for unmigrated user' do
+    teacher = create :teacher, :unmigrated_sso, provider: AuthenticationOption::CLEVER
+    sign_in teacher
+
+    expected_uri = "https://api.clever.com/v1.1/teachers/#{teacher.uid}/sections"
+    auth = {authorization: "Bearer #{teacher.oauth_token}"}
+    mock_response = {data: []}.to_json
+    RestClient.expects(:get).with(expected_uri, auth).returns(mock_response)
+    get :clever_classrooms
+  end
+
+  test 'clever_classrooms queries clever with clever authentication_id for migrated user' do
+    teacher = create :teacher, :with_migrated_clever_authentication_option
```

Comment:
It might be worth an `assert_nil teacher.uid` in the setup for this test to ensure incorrect behavior isn't masked by having the correct uid in two places.

Reply:
good call!